internal/terraform/graph_builder_plan.go

```diff
@@ -128,8 +128,16 @@ func (b *PlanGraphBuilder) Steps() []GraphTransformer {
 		},
 
 		// Add dynamic values
-		&RootVariableTransformer{Config: b.Config, RawValues: b.RootVariableValues, Planning: true},
-		&ModuleVariableTransformer{Config: b.Config, Planning: true},
+		&RootVariableTransformer{
+			Config:     b.Config,
+			RawValues:  b.RootVariableValues,
+			Planning:   true,
+			Destroying: b.Operation == walkPlanDestroy,
+		},
+		&ModuleVariableTransformer{
+			Config:   b.Config,
+			Planning: true,
```

Comment:
Should `Destroying` get set here instead, and then there are no checks to deal with during apply? This works as is, but it may be easier in the future if we can avoid registering `unknown` (or `invalid`, I can't remember the various states) checks in the first place.